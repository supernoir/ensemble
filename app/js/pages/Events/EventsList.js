import React from 'react';
import { Link } from 'react-router-dom';
import intl from 'react-intl-universal';
import { Container, Breadcrumb, Segment, Header, Feed, Divider, Icon } from 'semantic-ui-react';
import moment from 'moment';

export default class EventsList extends React.Component {
	parseEvent = event => {
		switch (event.action) {
			case 'add_project':
				return `You added ${event.ref} to your projects.`;
			case 'add_character':
				return `You added ${event.ref} to your Characters.`;
			case 'edit_project':
			case 'edit_character':
				return `You edited ${event.ref}`;
		}
	};

	componentDidMount() {
		this.props.getEvents();
	}

	render() {
		return (
			<Container>
				<Breadcrumb>
					<Breadcrumb.Section link>
						<Link to="/">{intl.get('component.dashboard')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section active>
						<Link to="/projects">{intl.get('entity.events')}</Link>
					</Breadcrumb.Section>
				</Breadcrumb>

				<Segment>
					<Header as="h2">
						{intl.get('entity.events')}
						<Icon name="newspaper outline"></Icon>
						<Header.Subheader>{intl.get('desc.events')}</Header.Subheader>
					</Header>
				</Segment>

				<Divider />

				{this.props.events
					? this.props.events.map(event => {
						return (
							<Feed>
								<Feed.Event>
									<Feed.Label image={'../../public/img/protouser.png'} />
									<Feed.Content>
										<Feed.Date>{moment(event.timestamp, 'X').format('lll')}</Feed.Date>
										<Feed.Summary>{this.parseEvent(event)}</Feed.Summary>
									</Feed.Content>
								</Feed.Event>
							</Feed>
						);
					  })
					: null}
			</Container>
		);
	}
}
