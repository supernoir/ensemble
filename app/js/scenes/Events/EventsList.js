import React from 'react';
import { Link } from 'react-router-dom';
import intl from 'react-intl-universal';
import { Container, Breadcrumb, Segment, Header, Feed, Divider } from 'semantic-ui-react';
import moment from 'moment';
import Loader from '../../basics/Loader';
import PropTypes from 'prop-types';

/**
 * Class EventsList
 * List of events
 */
export default class EventsList extends React.Component {
	parseEvent = event => {
		switch (event.action) {
			case 'add_project':
				return intl.get('event.action-addproject', { ref: event.ref });
			case 'add_character':
				return intl.get('event.action-addcharacter', { ref: event.ref });
			case 'edit_project':
				return intl.get('event.action-editproject', { ref: event.ref });
			case 'edit_character':
				return intl.get('event.action-editcharacter', { ref: event.ref });
			case 'delete_project':
				return intl.get('event.action-deletecharacter', { ref: event.ref });
			case 'delete_character':
				return intl.get('event.action-deletecharacter', { ref: event.ref });
			default: break;
		}
	};

	componentDidMount() {
		this.props.getEvents();
	}

	render() {
		return this.props.loading
			? <Loader loading={this.props.loading} />
			:	<Container>
				<Breadcrumb>
					<Breadcrumb.Section>
						<Link to="/">{intl.get('component.dashboard')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section active>
						<Link to="/projects">{intl.get('entity.events')}</Link>
					</Breadcrumb.Section>
				</Breadcrumb>

				<Segment>
					<Header as="h2">
						{intl.get('entity.events')}
						<Header.Subheader>{intl.get('desc.events')}</Header.Subheader>
					</Header>
				</Segment>

				<Divider />

				{this.props.events
					? this.props.events.map((event, index) => {
						return (
							<Feed key={`feed-${index}`}>
								<Feed.Event>
									<Feed.Label image={'../../public/img/protouser.png'} />
									<Feed.Content>
										<Feed.Date>{moment(event.timestamp, 'X').format('lll')}</Feed.Date>
										<Feed.Summary>{this.parseEvent(event)}</Feed.Summary>
									</Feed.Content>
								</Feed.Event>
							</Feed>
						);
					})
					: null}
			</Container>;
	}
}

EventsList.propTypes = {
	loading  : PropTypes.bool,
	getEvents: PropTypes.func,
	events   : PropTypes.arrayOf(
		PropTypes.shape({
			timestamp: PropTypes.timestamp,
			action   : PropTypes.string,
			ref      : PropTypes.string
		})
	)
};