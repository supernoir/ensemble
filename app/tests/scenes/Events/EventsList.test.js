import React from 'react';
import { shallow } from 'enzyme';

// Import Components
import EventsList from '../../../js/scenes/Events/EventsList';

test('EventsList -- Snapshot Test', () => {
	const wrapper = shallow(<EventsList getEvents={() => true} />);
	expect(wrapper).toMatchSnapshot();
});
