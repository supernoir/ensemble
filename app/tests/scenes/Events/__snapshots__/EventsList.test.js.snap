// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsList -- Snapshot Test 1`] = `
<Container>
  <Breadcrumb>
    <BreadcrumbSection
      link={true}
    >
      <Link
        replace={false}
        to="/"
      />
    </BreadcrumbSection>
    <BreadcrumbDivider />
    <BreadcrumbSection
      active={true}
    >
      <Link
        replace={false}
        to="/projects"
      />
    </BreadcrumbSection>
  </Breadcrumb>
  <Segment>
    <Header
      as="h2"
    >
      <HeaderSubheader />
    </Header>
  </Segment>
  <Divider />
</Container>
`;
